distributed/tests/test_worker.py

```diff
@@ -3502,3 +3505,43 @@ async def test_reconnect_argument_deprecated(s):
         warnings.simplefilter("error")
         async with Worker(s.address):
             pass
+
+
+@gen_test()
+async def test_worker_makes_own_thread():
+    class InitWorkerNewThread(WorkerPlugin):
```

Comment:
you should define the class outside the test so that you can use it without cloudpickle

Reply:
Is this in response to my comment above? If so, then I stopped getting the pickling error when I used popen to start the scheduler and worker rather than gen_cluster. But I'm happy to move the class outside the test if you prefer?

Reply:
yeah I think it would be better to use gen_cluster for this test

Reply:
I tried setting up the test as below, but I just get a timeout error on the client restart. I'm not sure why, seems to me that it should be equivalent to the popen method? 
```
class InitWorkerNewThread(WorkerPlugin):
    name: name
    thread: threading.Thread

    def __init__(self, scheduler):
        super().__init__()
        self.scheduler = scheduler

    def setup(self, worker):
        self.loop = asyncio.new_event_loop()
        self.thread = threading.Thread(target=self.loop.run_forever)
        self.thread.start()
        future = asyncio.run_coroutine_threadsafe(self.func(), self.loop)
        future.result()

    def teardown(self, worker):
        self.loop.call_soon_threadsafe(self.loop.stop)
        self.thread.join(timeout=10)

    async def func(self):
        try:
            worker = get_worker()
        except ValueError:
            async with Client(self.scheduler, asynchronous=True) as client:
                await client.register_worker_plugin(
                    InitWorkerNewThread(self.scheduler)
                )

@gen_cluster()
async def test_worker_makes_own_thread(s, a, b):
    async with Client(s.address, asynchronous=True) as c:
        await c.register_worker_plugin(InitWorkerNewThread(s.address))
        await c.restart()
        assert await c.submit(lambda x: x + 1, 1) == 2
```

Reply:
I just had a go at moving this out of the test definition, but it's not pickleable because `distributed/tests/__init__.py` is missing